File: src/app/types/application.ts
export type ApplicationType = "Merit" | "Social" | "Performance";
export type ApplicationStatus = "Draft" | "Pending Action" | "Approved";

export interface Application {
  id: number;
  type: ApplicationType;
  academicYear: string;
  semester: "I" | "II";
  createdAt: string;
  status: ApplicationStatus;
}

export interface CreateApplicationInput {
  type: ApplicationType;
  academicYear: string;
  semester: "I" | "II";
}

export interface UpdateApplicationInput {
  type?: ApplicationType;
  academicYear?: string;
  semester?: "I" | "II";
  status?: ApplicationStatus;
}

File: backend/src/main/java/com/formforge/model/Application.java
package com.formforge.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.PrePersist;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.format.DateTimeFormatter;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "applications")
public class Application {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ApplicationType type;

    @Column(name = "academic_year", nullable = false)
    private String academicYear;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Semester semester;

    @Column(name = "created_at")
    private String createdAt;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ApplicationStatus status;

    /** FK to the user who owns this application (nullable for legacy rows). */
    @Column(name = "owner_id")
    private Long ownerId;

    @PrePersist
    void prePersist() {
        if (createdAt == null) {
            createdAt = LocalDate.now().format(DateTimeFormatter.ofPattern("M/d/yyyy"));
        }
        if (status == null) {
            status = ApplicationStatus.DRAFT;
        }
    }
}
